Sources/Recycling/ListViews/FavoriteFolders/FavoriteFoldersListView.swift

```diff
@@ -274,23 +275,61 @@ extension FavoriteFoldersListView: UIScrollViewDelegate {
 extension FavoriteFoldersListView: UISearchBarDelegate {
     public func searchBarTextDidBeginEditing(_ searchBar: UISearchBar) {
         isSearchActive = true
-        footerView.isHidden = true
         searchBar.setShowsCancelButton(true, animated: true)
     }
 
     public func searchBarTextDidEndEditing(_ searchBar: UISearchBar) {
-        isSearchActive = false
-        footerView.isHidden = false
-        searchBar.setShowsCancelButton(false, animated: true)
-        delegate?.favoriteFoldersListViewDidCancelSearch(self)
+        // Hide the cancel button if the searchBar is empty.
+        // This will happen if the user removes the text and dismisses the keyboard by dragging.
+        //
+        // Otherwise enable the cancel button.
+        if searchBar.text?.isEmpty ?? true {
+            isSearchActive = false
+            searchBar.setShowsCancelButton(false, animated: true)
+        } else {
+            searchBar.enableCancelButton()
+        }
+    }
+
+    public func searchBarSearchButtonClicked(_ searchBar: UISearchBar) {
+        searchBar.resignFirstResponder()
     }
 
     public func searchBarCancelButtonClicked(_ searchBar: UISearchBar) {
+        isSearchActive = false
+
         searchBar.text = ""
-        searchBar.endEditing(false)
+        searchBar.setShowsCancelButton(false, animated: true)
+        searchBar.resignFirstResponder()
+
+        delegate?.favoriteFoldersListViewDidCancelSearch(self)
     }
 
     public func searchBar(_ searchBar: UISearchBar, textDidChange searchText: String) {
+        let showCancelButton = searchBar.isFirstResponder || !searchText.isEmpty
+        searchBar.setShowsCancelButton(showCancelButton, animated: true)
+        isSearchActive = !searchText.isEmpty
         delegate?.favoriteFoldersListView(self, didChangeSearchText: searchText)
     }
 }
+
+// MARK: - UISearchBar extension
+
+private extension UISearchBar {
+
+    /// The cancel button is automatically enabled/disabled based on the searchBar's firstResponder status.
+    /// We want to enable it when it's shown, no matter if the searchBar has focus or not.
+    func enableCancelButton() {
+        DispatchQueue.main.asyncAfter(deadline: .now() + 0.001, execute: { [weak self] in
```

Comment:
Why is this scheduled 1ms in the future?

Reply:
Yesterday the button was disabled if I only used `DispatchQueue.main.async`, but it worked if I gave it some delay.

Was going to make a gif showing this, but somehow it works now without delay. No clue why 🤔

Will remove the delay 👍

Reply:
Sounds to me like a fugly potential race, does it **always** ™️ work?

Reply:
Nevermind this one. We won't be showing the cancel button in the searchBar at all.